src/mochaVErrorReporter.ts

```diff
@@ -0,0 +1,47 @@
+import { Runner, utils as MochaUtils, reporters as MochaReporters } from "mocha";
+import { VError } from "verror";
+
+export class MochaVErrorReporter extends MochaReporters.Base
+{
+    private static _DEFAULT_INDENT: string = "    ";
+    private static _REMOVE_REPEATED_MESSAGE_REGEX: RegExp = /caused by.+\n/;
+    private static _REPLACE_WITH_INDENT_REGEX: RegExp = /\n/g;
+    private static _REMOVE_AFTER_FIRST_LINE_REGEX: RegExp = /\n.+/g;
+    private static stackFilter = MochaUtils.stackTraceFilter();
+
+    constructor(runner: Runner)
+    {
+        super(runner);
+        runner.on('fail', function(test, err) {
+            err.stack = MochaVErrorReporter.stackFilter(MochaVErrorReporter.fullStack(err));
+        });
+    }
+
+    public static fullStack(error, indent = MochaVErrorReporter._DEFAULT_INDENT, indentationLevel = 1)
```

Comment:
I've mentioned this before. Have a public method:
`public static fullStack(error, indent = MochaVErrorReporter._DEFAULT_INDENT)` and a private one that has the indentation level.

Reply:
Fixed after discussion